lib/chewy/strategy/atomic.rb

```diff
@@ -20,12 +20,20 @@ def update type, objects, options = {}
         ActiveSupport::Deprecation.warn("`urgent: true` option is deprecated and is not effective inside `:atomic` strategy, use `Chewy.strategy(:urgent)` strategy instead") if options.key?(:urgent)
 
         @stash[type] ||= []
-        @stash[type] |= type.adapter.identify(objects)
+        @stash[type] |= _identify_objects type, objects
       end
 
       def leave
         @stash.all? { |type, ids| type.import!(ids) }
       end
+
+      private
+
+      def _identify_objects type, objects
+        identifiers_from_evaluator = type.root_object.compose_id(objects) if type.root_object
+        identifiers_from_evaluator ||= nil
```

Comment:
Well - I checked via the spec that it is possible that if when the root_object is defined, but the _id evaluator is not defined resulted in `type.root_object.compose_id(objects)` returning nil. Mainly this was just a way of trying to deal with the else branch of the if at the end of the statement. 

So that when the if is false the variable would still end up being defined.